deps: bump cockroachdb/errors

knz opened the pull request:
This fixes a previously mistaken re-implementation of the
`errors.Cause()` API, intended for drop-in replacement with that of
`github.com/pkg/errors`.

I have verified that:

- none of the current code using `cockroachdb/errors` is currently
  relying on the (invalid) previous semantics.

- there was no important code that was previously upgraded from
  `pkg/errors` to `cockroachdb/errors` and using `errors.Cause`, and
  which could have been regressing without us realizing.

No important code, but still. There was exactly one instance of code
that regressed in such a way: the error check on the way out of the
CLI commands.

This aims to produce a helpful user message in various cases.  In the
special case of a gRPC deadline exceeded error the upgrade to
`cockroachdb/errors` broke that logic. This patch fixes that (resumes
the intended behavior).

Release note: None

knz committed:
deps: bump cockroachdb/errors

This fixes a previously mistaken re-implementation of the
`errors.Cause()` API, intended for drop-in replacement with that of
`github.com/pkg/errors`.

I have verified that:

- none of the current code using `cockroachdb/errors` is currently
  relying on the (invalid) previous semantics.

- there was no important code that was previously upgraded from
  `pkg/errors` to `cockroachdb/errors` and using `errors.Cause`, and
  which could have been regressing without us realizing.

No important code, but still. There was exactly one instance of code
that regressed in such a way: the error check on the way out of the
CLI commands.

This aims to produce a helpful user message in various cases.  In the
special case of a gRPC deadline exceeded error the upgrade to
`cockroachdb/errors` broke that logic. This patch fixes that (resumes
the intended behavior).

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/42866)
<!-- Reviewable:end -->

knz committed:
deps: bump datadriven

This fixes a bug where datadriven would ignore all the files in
a directory following an ignored file.

Release note: none

knz commented:
TFYR!

bors r+
